src/components/TimetableV2.tsx

```diff
@@ -0,0 +1,1176 @@
+    const scheduledDateTimes = scheduledPlannables.map(
+      (program): [Temporal.PlainDateTime, Temporal.Duration] => [
+        Temporal.Instant.fromEpochMilliseconds(program.begin)
+          .toZonedDateTimeISO(LOCAL_TIMEZONE)
+          .toPlainDateTime(),
+        Temporal.Duration.from({ milliseconds: program.duration }),
+      ],
+    );
+
+    // Find the first and last date of the schedule
+    let [firstDate, lastDate] = scheduledDateTimes.reduce<
+      [Temporal.PlainDate | null, Temporal.PlainDate | null]
+    >(
+      ([firstDate, lastDate], [start, duration]) => {
+        const startPlain = start.toPlainDate();
+        const endPlain = start.add(duration).toPlainDate();
+        return [
+          firstDate === null ||
+          Temporal.PlainDate.compare(startPlain, firstDate) < 0
+            ? startPlain
+            : firstDate,
+          lastDate === null ||
+          Temporal.PlainDate.compare(endPlain, lastDate) > 0
+            ? endPlain
+            : lastDate,
+        ];
+      },
+      [null, null],
+    );
+
+    // If there are no scheduled programs, show only the current date
+    if (firstDate === null || lastDate === null) {
+      return [Temporal.Now.plainDateISO(LOCAL_TIMEZONE)];
+    }
+
+    // Start one day before the first date and end one day after the last date
+    firstDate = firstDate.subtract({ days: 1 });
+    lastDate = lastDate.add({ days: 1 });
+
+    // Get all dates between the first and last date
+    let last = firstDate;
+    const dates = [last];
+    while (Temporal.PlainDateTime.compare(last, lastDate) < 0) {
+      last = last.add(Temporal.Duration.from({ days: 1 }));
+      dates.push(last);
+    }
+    return dates;
+  }, [scheduledPlannables]);
+
+  const atendeeGroups = useGroups();
+
+  // Show virtual group on the top, if there are no groups or if there are programs without groups
+  const hasAtLeastOneGroup = atendeeGroups.length > 0;
+  const programsWithoutGroups = scheduledPlannables.filter(
+    (it) => it.groups.length === 0,
+  );
+  const showVirtualGroup =
+    !hasAtLeastOneGroup || programsWithoutGroups.length > 0;
+  const shownGroups: string[] | [null, ...string[]] = useMemo(() => {
+    const sortedAttendeGroupIds = atendeeGroups.map((it) => it._id);
+    return showVirtualGroup
+      ? [null, ...sortedAttendeGroupIds]
+      : sortedAttendeGroupIds;
+  }, [atendeeGroups, showVirtualGroup]);
+
+  // Owner filtering
+  const [ownerFilter, setOwnerFilter] = useState<string | null>(null);
+  const people = usePeople();
+  const availableOwners = useMemo(() => {
+    const result: { id: string; name: string; count: number }[] = [];
+    for (const program of programs) {
+      for (const { person: ownerId } of program.people) {
+        let ownerRecord = result.find((it) => it.id === ownerId);
+        if (!ownerRecord) {
+          const person = people.find((it) => it._id === ownerId);
+          ownerRecord = { id: ownerId, name: person!.name, count: 0 };
```

Comment:
Looks like this is failing due to data inconsistency. There is no guarantee that `people.find(...)` will succeed, so there needs to be some fallback in `person!.name`.